test/suites/consolidation/suite_test.go

```diff
@@ -98,40 +164,99 @@ var _ = Describe("Consolidation", func() {
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("4")},
+				},
+			},
+		})
+		smallDep := test.Deployment(test.DeploymentOptions{
+			Replicas: numPods,
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "small-app"},
+				},
+				TopologySpreadConstraints: []v1.TopologySpreadConstraint{
+					{
+						MaxSkew:           1,
+						TopologyKey:       v1.LabelHostname,
+						WhenUnsatisfiable: v1.DoNotSchedule,
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"app": "small-app",
+							},
+						},
+					},
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1.5")},
 				},
 			},
 		})
 
-		selector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+		selector := labels.SelectorFromSet(largeDep.Spec.Selector.MatchLabels)
 		env.ExpectCreatedNodeCount("==", 0)
-		env.ExpectCreated(nodePool, nodeClass, dep)
+		env.ExpectCreated(nodePool, nodeClass, largeDep, smallDep)
 
 		env.EventuallyExpectHealthyPodCount(selector, int(numPods))
 
-		// reduce the number of pods by 60%
-		dep.Spec.Replicas = aws.Int32(40)
-		env.ExpectUpdated(dep)
+		// 3 nodes due to the anti-affinity rules
+		env.ExpectCreatedNodeCount("==", 3)
+
+		// scaling down the large deployment leaves only small pods on each node
+		largeDep.Spec.Replicas = aws.Int32(0)
+		env.ExpectUpdated(largeDep)
 		env.EventuallyExpectAvgUtilization(v1.ResourceCPU, "<", 0.5)
 
-		// Enable consolidation as WhenUnderutilized doesn't allow a consolidateAfter value
 		nodePool.Spec.Disruption.ConsolidateAfter = nil
 		env.ExpectUpdated(nodePool)
 
-		// With consolidation enabled, we now must delete nodes
-		env.EventuallyExpectAvgUtilization(v1.ResourceCPU, ">", 0.6)
+		// With consolidation enabled, we now must replace each node in turn to consolidate due to the anti-affinity
+		// rules on the smaller deployment.  The 2xl nodes should go to a large
+		env.EventuallyExpectAvgUtilization(v1.ResourceCPU, ">", 0.8)
 
-		env.ExpectDeleted(dep)
+		var nodes v1.NodeList
+		Expect(env.Client.List(env.Context, &nodes)).To(Succeed())
+		numLargeNodes := 0
+		numOtherNodes := 0
+		for _, n := range nodes.Items {
+			// only count the nodes created by the provisoiner
+			if n.Labels[corev1beta1.NodePoolLabelKey] != nodePool.Name {
+				continue
+			}
+			if strings.HasSuffix(n.Labels[v1.LabelInstanceTypeStable], ".large") {
+				numLargeNodes++
+			} else {
+				numOtherNodes++
+			}
+		}
+
+		// all of the 2xlarge nodes should have been replaced with large instance types
+		Expect(numLargeNodes).To(Equal(3))
+		// and we should have no other nodes
+		Expect(numOtherNodes).To(Equal(0))
+
+		env.ExpectDeleted(largeDep, smallDep)
 	})
-	It("should consolidate on-demand nodes (replace)", func() {
+	It("should consolidate spot nodes (replace)", func() {
```

Comment:
can we collapse this test and the one above with a DescribeTable or are they different enough that they need to stay separated?